### PR TITLE
docs: add automatic global registration to cookbook

### PR DESCRIPTION
**Pull request by vilaboim**
## Description of Problem

In #43 there is a proposal by @bencodezen to move "### Automatic Global Registration of Base Components" to Cookbook.

## Proposed Solution

I updated the Vue 2 example and added a page inside Cookbook.

## Additional Information
Related to #565

### CODE REVIEW

File: src/cookbook/automatic-global-registration-of-base-components.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+Just to support relatively little markup in a template:
+
+```html
+<BaseInput v-model="searchText" v-on:keydown.enter="search" />
+<BaseButton v-on:click="search">
+  <BaseIcon name="search" />
+</BaseButton>
+```
+
+Fortunately, if you're using Webpack (or [Vue CLI](https://github.com/vuejs/vue-cli), which uses Webpack internally), you can use `require.context` to globally register only these very common base components. Here's an example of the code you might use to globally import base components in your app's entry file (e.g. `src/main.js`):
```
Comment (NataliaTepluhina) [inline src/cookbook/automatic-global-registration-of-base-components.md:31]: **suggestion:**  I think we should use `webpack` lowercased everywhere as it's [officially lowercased](https://twitter.com/wsokra/status/855800490713649152?lang=en). I know we should fix this across all the docs but for now let's not add more capitalized Webpack to the codebase 😄

**Review by NataliaTepluhina — COMMENTED** [review]
@vilaboim great addition to the cookbook! I've left a few nitpicky comments here and there, rest looks good to me :+1:

File: src/cookbook/automatic-global-registration-of-base-components.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# Automatic Global Registration of Base Components
+
+## Base Example
+
+Many of your components will be relatively generic, possibly only wrapping an element like an input or a button. We sometimes refer to these as [base components](../style-guide/#Base-component-names-strongly-recommended) and they tend to be used very frequently across your components.
```
Comment (skirtles-code) [inline src/cookbook/automatic-global-registration-of-base-components.md:5]:
```suggestion
Many of your components will be relatively generic, possibly only wrapping an element like an input or a button. We sometimes refer to these as [base components](../style-guide/#base-component-names-strongly-recommended) and they tend to be used very frequently across your components.
```

**Review by skirtles-code — CHANGES_REQUESTED** [review]
I've suggested a tweak to the link so that it jumps to the relevant section.

File: src/cookbook/automatic-global-registration-of-base-components.md (hunk 1 above)
Comment (vilaboim) [reply src/cookbook/automatic-global-registration-of-base-components.md:31]: I hadn't realized that. 😄

File: src/cookbook/automatic-global-registration-of-base-components.md (hunk 2 above)
Comment (vilaboim) [reply src/cookbook/automatic-global-registration-of-base-components.md:5]: Thank you @skirtles-code, great observation.

**Pull request merged**